internal/services/machinelearning/machine_learning_workspace_hub_resource.go

```diff
@@ -0,0 +1,496 @@
+	Identity                    []identity.ModelSystemAssignedUserAssigned `tfschema:"identity"`
+	Encryption                  []encryptionModel                          `tfschema:"encryption"`
+	Tags                        map[string]string                          `tfschema:"tags"`
+}
+
+type encryptionModel struct {
+	KeyVaultID             string `tfschema:"key_vault_id"`
+	KeyID                  string `tfschema:"key_id"`
+	UserAssignedIdentityID string `tfschema:"user_assigned_identity_id"`
+}
+
+type MachineLearningWorkspaceHubResource struct{}
+
+var _ sdk.Resource = MachineLearningWorkspaceHubResource{}
+
+func (r MachineLearningWorkspaceHubResource) ResourceType() string {
+	return "azurerm_machine_learning_workspace_hub"
+}
+
+func (r MachineLearningWorkspaceHubResource) ModelObject() interface{} {
+	return &machineLearningWorkspaceHubModel{}
+}
+
+func (r MachineLearningWorkspaceHubResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return workspaces.ValidateWorkspaceID
+}
+
+func (r MachineLearningWorkspaceHubResource) Arguments() map[string]*pluginsdk.Schema {
+	arguments := map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"location": commonschema.Location(),
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"description": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+		},
+
+		"key_vault_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: commonids.ValidateKeyVaultID,
+		},
+
+		"storage_account_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: commonids.ValidateStorageAccountID,
+		},
+
+		"application_insights_id": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ForceNew:     true,
+			ValidateFunc: components.ValidateComponentID,
+		},
+
+		"identity": commonschema.SystemAssignedUserAssignedIdentityRequired(),
+
+		"container_registry_id": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ForceNew:     true,
+			ValidateFunc: registries.ValidateRegistryID,
+		},
+
+		"encryption": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			ForceNew: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"key_vault_id": commonschema.ResourceIDReferenceRequired(&commonids.KeyVaultId{}),
+					"key_id": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.IsURLWithHTTPorHTTPS,
+					},
+					"user_assigned_identity_id": {
+						Type:             pluginsdk.TypeString,
+						Optional:         true,
+						ValidateFunc:     commonids.ValidateUserAssignedIdentityID,
+						DiffSuppressFunc: suppress.CaseDifference,
```

Comment:
how come this is required? if it is that is likely a bug in the API - could you open an issue on the rest epcs and link it here?

Reply:
will check this with service team. As far as I know this issue has not been fixed and we need to suppress case insensitive diff